sklearn/utils/validation.py

```diff
@@ -980,3 +980,60 @@ def check_scalar(x, name, target_type, min_val=None, max_val=None):
 
     if max_val is not None and x > max_val:
         raise ValueError('`{}`= {}, must be <= {}.'.format(name, x, max_val))
+
+
+def _check_sample_weight(sample_weight, y=None, n_samples=None, dtype=None,
+                         order=None, **kwargs):
```

Comment:
some estimators (e.g. AdaBoost) might need the weights to be normalized (i.e. `w / w.sum()`). You could add a normalize parameter ?

Reply:
Or do we let each estimator do it since that it might be on estimator bases?

Reply:
Yeah, for now the approach was to add functionality as I encountered it in the code. `AdaBoost` seems to do something fairly specific, I haven't seen the need for it in other estimators. But if we encounter more, we could certainly add it. This would need to be applied to more estimators in the future (and that's also why I would rather not make the function public, as parameters may change)..

Reply:
I agree to make it private until all estimators use it and the functionalities are stable.

Reply:
It seem to be specific to AdaBoost. Kmeans uses a different normalization (weights have to sum to n_samples...)